### ISSUE TITLE
Backtest ends before the predefined range in commands

### ISSUE THREAD

**Issue by qhungbui7** [freqtrade/freqtrade#10479]
Hi, 
I am currently running this command and freqtrade returns shorter results in the timerange. 
freqtrade backtesting --timerange 20240709-20240721 --timeframe 1m --strategy BacktestAdapter --cache none

![image](https://github.com/user-attachments/assets/efc920ac-71c0-4c17-969c-62ae4f7ba7a9)

About the data, I only download a certain range of the data when my model decides to trade to reduce the cost of storage. In this case, there is a trade in 2024-07-18T13:00:13.000Z and the fetched dataframe from freqtrade does not extend to that moment. I already use this command to download that part of the data but I noticed that the log shows that it downloaded a different part of the data. 

![image](https://github.com/user-attachments/assets/24c829af-e72d-41dd-bcb3-612c0c0b9df6)

So, I don't know if this is a bugs or Freqtrade simply does not support this idea. Thank you for reading.

**Comment by xmatthias** [comment]
Please get used to filling out the issue template.  Those who frequently omit the template may have their issues closed without comment.
We're asking some very basic questions - by deleting the issue template, you're explicitly choosing not to answer our questions.

Filling out the template helps reduce our workload, and we greatly appreciate that.

---

Please provide
* logs as text, not screenshots
* Full backtest logs
* Filled out issue template (treating this as outdated version until this has been completed).

**Comment by xmatthias** [comment]
Reading your comment more carefully - the answer is no - freqtrade does not support what you're asking (and we also don't plan on supporting this).

Backtesting happens for the given timerange - with the assumption that **all data is available**.
Backtesting will however also stop when the data ends (at both sides) (hence your output stops before the given timerange - as your data ends on 2024-07-14 (the date given in the backtesting output)). 
It also assumes that the data is **complete with no holes** - and commands like `download-data` will try hard to ensure that the data is complete and without data gaps.


your download-data command suffers different problems. 
you use `--prepend` - which means it'll **only** prepend to existing data. Your existing data starts before the given start time - (20240717-20240720 - data starts at 2024-07-10 - so there's nothing to prepend as data starts before your given range.
You'll most likely want to remove `--prepend` - which is the default and will append data to the end if necessary. It's however a "either or" flag - if it's given - data download will always stop at the existing data start time.

simply put - to use `freqtrade backtesting --timerange 20240709-20240721 ...` - freqtrade requires all data between these 2 dates to be present - + a bit more before to account for startup_candle_count.

**Comment by qhungbui7** [comment]
Sorry for the inconvenience. I understand, thank you for your time. Appreciating your work.
